### PR TITLE
Fix `deny(invalid_doc_attributes)`

### LINKED ISSUE #85497: invalid_doc_attributes cannot be allowed or denied
I tried this code:

```rust
#![deny(invalid_doc_attributes)]
#![doc(x)]
```

I expected to see this happen: An error that `x` is invalid.

Instead, this happened: The lint stays at a warning and rustc additionally warns that the lint name isn't recognized (https://github.com/rust-lang/rust/pull/85457#issuecomment-844446870):

```
warning: unknown lint: `invalid_doc_attributes`
 --> src/lib.rs:1:9
  |
1 | #![deny(invalid_doc_attributes)]
  |         ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unknown_lints)]` on by default

warning: unknown `doc` attribute `x`
 --> src/lib.rs:2:8
  |
2 | #![doc(x)]
  |        ^
  |
  = note: `#[warn(invalid_doc_attributes)]` on by default
  = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #82730 <https://github.com/rust-lang/rust/issues/82730>
```

### Meta

`rustc --version --verbose`:  1.54.0-nightly (2021-05-18 4e3e6db011c5b482d2be)

### PR DESCRIPTION
**Pull request by jyn514**
Fixes https://github.com/rust-lang/rust/issues/85497.

### CODE REVIEW

File: src/test/ui/rustdoc/deny-invalid-doc-attrs.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#![feature(no_core)]
+#![no_core]
```
Comment (jyn514) [inline src/test/ui/rustdoc/deny-invalid-doc-attrs.rs:2]: Oh oops, this isn't strictly necessary, I just am unable to build libstd locally due to https://github.com/rust-lang/rust/issues/56650.

**Comment by camelid** [comment]
Does this fix it for it being implicitly denied via `deny(warnings)` too?

**Comment by jyn514** [comment]
> Does this fix it for it being implicitly denied via `deny(warnings)` too?

Yes, I added a test case for that (as part of `doc-spotlight.rs`).

**Pull request merged**